third-party/cmake/GetWindowsDeps.cmake

```diff
@@ -0,0 +1,49 @@
+# Download and install binary dependencies for windows
+include(CMakeParseArguments)
+
+# This is similar to the build recipes, but instead downloads a third party
+# binary and installs it under the the DEPS_PREFIX.
+function(GetWindowsDep)
+  cmake_parse_arguments(_gettool
+    "BUILD_IN_SOURCE"
+    "TARGET"
+    "INSTALL_COMMAND"
+    ${ARGN})
+
+  if(NOT _gettool_TARGET OR NOT _gettool_INSTALL_COMMAND)
+    message(FATAL_ERROR "Must pass INSTALL_COMMAND and TARGET")
+  endif()
+
+  string(TOUPPER "${_gettool_TARGET}_URL" URL_VNAME)
+  string(TOUPPER "${_gettool_TARGET}_SHA256" HASH_VNAME)
+  set(URL ${${URL_VNAME}})
+  set(HASH ${${HASH_VNAME}})
+  if(NOT URL OR NOT HASH )
+	  message(FATAL_ERROR "${URL_VNAME} and ${HASH_VNAME} must be set")
+  endif()
+
+  ExternalProject_Add(${_gettool_TARGET}
+    PREFIX ${DEPS_BUILD_DIR}
+    URL ${URL}
+    DOWNLOAD_DIR ${DEPS_DOWNLOAD_DIR}
+    DOWNLOAD_COMMAND ${CMAKE_COMMAND}
+      -DPREFIX=${DEPS_BUILD_DIR}
+      -DDOWNLOAD_DIR=${DEPS_DOWNLOAD_DIR}
+      -DURL=${URL}
+      -DEXPECTED_SHA256=${HASH}
+      -DTARGET=${_gettool_TARGET}
+      -DUSE_EXISTING_SRC_DIR=${USE_EXISTING_SRC_DIR}
+      -P ${CMAKE_CURRENT_SOURCE_DIR}/cmake/DownloadAndExtractFile.cmake
+    CONFIGURE_COMMAND ""
+    BUILD_IN_SOURCE 1
+    CONFIGURE_COMMAND ""
+    BUILD_COMMAND ""
+    INSTALL_COMMAND "${_gettool_INSTALL_COMMAND}")
+  list(APPEND THIRD_PARTY_DEPS ${__gettool_TARGET})
+endfunction()
+
+GetWindowsDep(TARGET win32yank
+  INSTALL_COMMAND ${CMAKE_COMMAND} -E make_directory ${DEPS_INSTALL_DIR}/bin
+    COMMAND ${CMAKE_COMMAND} -E copy 
+             ${DEPS_BUILD_DIR}/src/win32yank/win32yank.exe
+	     ${DEPS_INSTALL_DIR}/bin)
```

Comment:
Can this logic be part of `GetWindowsDep()`? Then in `third-party/CMakeLists.txt`, after `include(GetWindowsDeps)`, we could call `GetWindowsDep()` there instead of here. That avoids have "win32yank" specified in three different places. That will make things a bit more maintainable as we add more "binary dependencies".